File: problemas/1000/1051.dart
import 'dart:io';

void main() {
  double salario = double.parse(stdin.readLineSync());
  double imposto = 0.0;

  var faixas = {
    4500.00: 0.28,
    3000.00: 0.18,
    2000.00: 0.08
  };

  faixas.forEach((valor, aliquota) {
    if (salario > valor) {
      double diferenca = salario - valor;
      imposto += diferenca * aliquota;
      salario -= diferenca;
    }
  });

  stdout.writeln(imposto == 0.0 ? 'Isento' : 'R\$ ${imposto.toStringAsFixed(2)}');
}

File: problemas/1000/1051.py
salario = float(input())
imposto = 0.0

faixas = [
    (4500.00, 0.28),
    (3000.00, 0.18),
    (2000.00, 0.08)
]

for (valor, aliquota) in faixas:
    if salario > valor:
        diferenca = salario - valor
        imposto += diferenca * aliquota
        salario -= diferenca

print(f'Isento' if imposto == 0.0 else 'R$ {imposto:.2f}')
